src/main/kotlin/no/nav/syfo/narmestelederrelasjon/database/NarmesteLederRelasjonQuery.kt

```diff
@@ -0,0 +1,104 @@
+import java.sql.*
+import java.time.Instant
+import java.time.ZoneOffset
+import java.util.*
+
+const val queryCreateNarmesteLederRelasjon =
+    """
+    INSERT INTO NARMESTE_LEDER_RELASJON (
+        id,
+        uuid,
+        created_at,
+        updated_at,
+        referanse_uuid,
+        virksomhetsnummer,
+        arbeidstaker_personident,
+        narmeste_leder_personident,
+        narmeste_leder_telefonnummer,
+        narmeste_leder_epost,
+        arbeidsgiver_forskutterer,
+        aktiv_fom,
+        aktiv_tom,
+        timestamp
+        ) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) RETURNING id
+    """
+
+fun Connection.createNarmesteLederRelasjon(
+    commit: Boolean = true,
+    narmesteLederLeesah: NarmesteLederLeesah,
+) {
+    val now = Timestamp.from(Instant.now())
+
+    val narmesteLederRelasjonUuid = UUID.randomUUID()
+
+    val narmesteLederRelasjonIdList = this.prepareStatement(queryCreateNarmesteLederRelasjon).use {
+        it.setString(1, narmesteLederRelasjonUuid.toString())
+        it.setTimestamp(2, now)
+        it.setTimestamp(3, now)
+        it.setString(4, narmesteLederLeesah.narmesteLederId.toString())
+        it.setString(5, narmesteLederLeesah.orgnummer)
+        it.setString(6, narmesteLederLeesah.fnr)
+        it.setString(7, narmesteLederLeesah.narmesteLederFnr)
+        it.setString(8, narmesteLederLeesah.narmesteLederTelefonnummer)
+        it.setString(9, narmesteLederLeesah.narmesteLederEpost)
+        it.setObject(10, narmesteLederLeesah.arbeidsgiverForskutterer)
+        it.setTimestamp(11, Timestamp.valueOf(narmesteLederLeesah.aktivFom.atStartOfDay()))
+        it.setTimestamp(12, narmesteLederLeesah.aktivTom?.let { aktivTom -> Timestamp.valueOf(aktivTom.atStartOfDay()) })
+        it.setTimestamp(13, Timestamp.from(narmesteLederLeesah.timestamp.toInstant()))
+        it.executeQuery().toList { getInt("id") }
+    }
+
+    if (narmesteLederRelasjonIdList.size != 1) {
+        throw SQLException("Creating MoteStatusEndring failed, no rows affected.")
+    }
+
+    if (commit) {
+        this.commit()
+    }
+}
+
+const val queryGetNarmesteLederRelasjonList =
+    """
+    SELECT *
+    FROM narmeste_leder_relasjon
+    WHERE arbeidstaker_personident = ?;
+    """
+
+fun DatabaseInterface.getNarmesteLederRelasjonList(
+    personIdentNumber: PersonIdentNumber,
+): List<PNarmesteLederRelasjon> {
+    return this.connection.use { connection ->
+        connection.prepareStatement(queryGetNarmesteLederRelasjonList).use {
+            it.setString(1, personIdentNumber.value)
+            it.executeQuery().toList {
+                toPNarmesteLederRelasjon()
+            }
+        }
+    }
+}
+
+fun ResultSet.toPNarmesteLederRelasjon(): PNarmesteLederRelasjon =
+    PNarmesteLederRelasjon(
+        id = getInt("id"),
+        uuid = UUID.fromString(getString("uuid")),
+        createdAt = getTimestamp("created_at").toInstant().atOffset(ZoneOffset.UTC),
```

Comment:
Tror du må lage en test på at du faktisk får tilbake samme tidspunkt som ble skrevet til databasen. Jeg tror dette gjør at tidspunktet drifter en time eller to (avhengig av sommertid eller ikke) fordi du skriver det til databasen med tidssone, men leser tilbake utc-verdien (men dette er litt forvirrende så kanskje jeg tar feil 😉)

Reply:
Det var feil her ja.
Nå har jeg endret det slik at alle tidspunkter inn og ut i databasen bruker offset UTC (`val defaultZoneOffset: ZoneOffset = ZoneOffset.UTC`)